### PR TITLE
[Bug] handle null state for SkillLevel on screening decision dialog

### LINKED ISSUE #9757: 🐛 Null state for pool skill levels on the assessment screening dialog is not very user friendly
## 🐛 Bug
When we roll out skill levels for pool creation , all existing assessments will show the assessment screening dialog as below : 
Screenshot 2024-03-14 161700

side effect on #9658 
## 🦋 Expected Behaviour

When there is no skill levels leave it blank. Do not add 'Not found'. It doesn't work in this context

## 🕵️ Details

Add any additional details that could assist with troubleshooting/fixing the issue.

- **Operating System**: Name and version (if possible).
- **Browser**: Name and version (if possible).

## 📋 Steps to Reproduce

1. Open up assessment screening dialog on any existing pools 

## 📸 Screenshot

Add a screenshot (if possible).

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] In this scenario, just omit skill level on the title and the definition section

### PR DESCRIPTION
**Pull request by brindasasi**
🤖 Resolves #9757 

## 👋 Introduction

This PR omits `Not Found`  key word when there is no required level for the skill on the screening decision dialog title and inside the description section 

## 🕵️ Details

Add any additional details that could assist with reviewing or testing this PR.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Open any existing process without required level on the skills ( make sure candidates exist)
2. Go to screening and assessment tab 
3. Click on `To assess` for any candidate to open up the screening decision dialog
4. Verify `Not Found ` doesn't appear in the dialog 

Regression Test
1. Open any existing process WITH required level on the skills ( make sure candidates exist)
2. Go to screening and assessment tab 
3. Click on `To assess` for any candidate to open up the screening decision dialog
4. Verify Skill Levels are appearing properly for the Title and the description of the dialog.

## 📸 Screenshot

When there is no required level 
Screenshot 2024-03-18 at 10 22 41 AM
When there is required level 
Screenshot 2024-03-18 at 2 01 25 PM

### CODE REVIEW

File: apps/web/src/lang/fr.json [hunk 1]
```diff
@@ -6606,6 +6606,10 @@
     "defaultMessage": "Ajouter une nouvelle compétence",
     "description": "Button text to open the skill dialog and add a skill"
   },
+  "ZZpC8s": {
+    "defaultMessage": "Voir les définitions pour \"{skillName}\"",
```
Comment (mnigh) [inline apps/web/src/lang/fr.json:6610]:
```suggestion
    "defaultMessage": "Voir les définitions pour « {skillName} »",
```

File: apps/web/src/lang/fr.json [hunk 2]
```diff
@@ -7166,6 +7170,10 @@
     "defaultMessage": "Le gouvernement du Canada se réserve le droit de supprimer des commentaires et des contributions, et de bloquer des utilisateurs en fonction des critères ci-dessous:",
     "description": "Paragraph for comments and interaction section"
   },
+  "cnoxKB": {
+    "defaultMessage": "Examiner et consigner les résultats de {candidateName} concernant \"{skillName}\".",
```
Comment (mnigh) [inline apps/web/src/lang/fr.json:7174]:
```suggestion
    "defaultMessage": "Examiner et consigner les résultats de {candidateName} concernant « {skillName} ».",
```

**Review by mnigh — DISMISSED** [review]

**Review by esizer — COMMENTED** [review]
Tried reviewing this but actually got a 500 from the API :cry: 

screenshot_19-Mar-2024_08-28-1710851337

**Comment by brindasasi** [comment]
> Tried reviewing this but actually got a 500 from the API 😢
> 
> screenshot_19-Mar-2024_08-28-1710851337

how did you get this error ? 🤔  It works fine for me

**Comment by brindasasi** [comment]
> > how did you get this error ? 🤔 It works fine for me
> 
> I just updated a pool to not have `required_level` like the first step suggested 😅
> 
> > Open any existing process without required level on the skills

I'm able to reproduce the error when I wipe out the existing required level and replace it with empty string. 
The reason is that `PoolCandidateSnapshot` query couldn't parse the empty string into enum. 
I don't think anyone would be able to add empty string for the Required Level via UI and not a valid use case IMO.

Currently in prod all existing processes have NULL in their required column. 
When I wipe out the required level column and leave it as NULL, the fix works fine.

Let me know what do you think on that.

**Comment by vd1992** [comment]
Works... Partially

image

File: apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx [hunk 3]
```diff
@@ -74,6 +75,38 @@ const getSkillLevelMessage = (
   return skillLevel;
 };
 
+const getTitle = (poolSkill: PoolSkill | undefined, intl: IntlShape) => {
+  let title = "";
+  const skillLevel = getSkillLevelMessage(poolSkill, intl);
+  if (!isEmpty(skillLevel)) {
```
Comment (vd1992) [inline apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx:81]: `getSkillLevelMessage()` will never fail the check as it returns a truthy string guaranteed, it either returns a skill level or the fallback `notFound`

image

Comment (brindasasi) [reply apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx:81]: Oops ..looks like I have put back the old code during merge conflict.

**Review by vd1992 — APPROVED** [review]
Looks good!

**Review by esizer — APPROVED** [review]
LGTM! :+1:

**Pull request merged**